static/js/tools/map/bq_query_utils.test.ts

```diff
@@ -0,0 +1,485 @@
+            EXISTS(SELECT * FROM UNNEST(linked_contained_in_place) AS C WHERE C = 'geoId/06')
+    )
+    SELECT O.variable_measured as StatVarId,
+           O.observation_about as PlaceId,
+           MAX(O.Observation_date) as LatestDate
+    FROM \`data_commons.Observation\` as O
+    JOIN ChildPlace ON TRUE
+    JOIN \`data_commons.Provenance\` as I ON TRUE
+    WHERE O.prov_id = I.prov_id AND
+          O.variable_measured = 'Count_Person_Female' AND
+          O.observation_about = ChildPlace.PlaceId AND
+          O.measurement_method = 'testMMethod' AND
+          O.unit IS NULL AND
+          O.observation_period = 'P1Y' AND
+          O.scaling_factor IS NULL AND
+          I.name IS NULL
+    GROUP BY StatVarId, PlaceId
+  )
+  SELECT ONum.observation_about AS PlaceId,
+         ONum.variable_measured AS NumVariableId,
+         ONum.observation_date AS NumDate,
+         ODenom.variable_measured AS DenomVariableId,
+         ODenom.observation_date AS DenomDate,
+         MIN(ODenom.facet_rank) AS DenomRank
+  FROM \`data_commons.Observation\` AS ONum
+  JOIN \`data_commons.Observation\` AS ODenom ON TRUE
+  JOIN LatestObsDate ON TRUE
+  JOIN \`data_commons.Provenance\` AS I ON TRUE
+  WHERE ONum.observation_about = LatestObsDate.PlaceId AND
+        ONum.variable_measured = 'Count_Person_Female' AND
+        ONum.prov_id = I.prov_id AND
+        ODenom.observation_about = ONum.observation_about AND
+        ODenom.variable_measured = '' AND
+        ODenom.observation_date = SUBSTR(ONum.observation_date, 0, 4) AND
+        ONum.observation_date = LatestObsDate.LatestDate AND
+        ONum.measurement_method = 'testMMethod' AND
+        ONum.unit IS NULL AND
+        ONum.observation_period = 'P1Y' AND
+        ONum.scaling_factor IS NULL AND
+        I.name IS NULL
+  GROUP BY PlaceId, NumVariableId, NumDate, DenomVariableId, DenomDate
+)
+SELECT ONum.observation_about AS PlaceId,
+      P.name AS PlaceName,
+      ONum.observation_date AS LatestDate,
+      ONum.measurement_method AS MeasurementMethod,
+      ONum.unit AS Unit,
+      NET.REG_DOMAIN(I.provenance_url) AS Source,
+      IF(ODenom.value IS NOT NULL AND CAST(ODenom.value AS FLOAT64) > 0,
+        CAST(ONum.value AS FLOAT64) / CAST(ODenom.value AS FLOAT64),
+        NULL) AS ValuePerCapita,
+      ODenom.value AS DenomValue,
+      ODenom.observation_date AS DenomDate,
+FROM \`data_commons.Observation\` AS ONum
+JOIN \`data_commons.Place\` AS P ON TRUE
+JOIN \`data_commons.Provenance\` AS I ON TRUE
+JOIN \`data_commons.Observation\` AS ODenom ON TRUE
+JOIN PlaceObsDatesAndDenomRank AS PODDR ON TRUE
+WHERE ONum.variable_measured = 'Count_Person_Female' AND
+      ONum.observation_about = PODDR.PlaceId AND
+      ONum.observation_about = P.id AND
+      ONum.prov_id = I.id AND
+      ONum.measurement_method = 'testMMethod' AND
+      ONum.unit IS NULL AND
+      ONum.observation_period = 'P1Y' AND
+      ONum.scaling_factor IS NULL AND
+      I.name IS NULL AND
```

Comment:
I don't think we should do this.  If importNAme is not provided, we should not include it in the join.